Sources/PathKit.swift

```diff
@@ -152,8 +152,18 @@ extension Path {
   ///
   public func abbreviate() -> Path {
 #if os(Linux)
-    // TODO: actually de-normalize the path
-    return self
+    let env = ProcessInfo.processInfo.environment
+    guard let home = env["HOME"], !home.isEmpty else { return self }
+    guard let homeRange = self.path.range(of: home, options: [.anchored, .caseInsensitive]) else { return self }
+    let withoutHome = Path(self.path.replacingCharacters(in: homeRange, with: ""))
+    
+    if withoutHome.path.isEmpty || withoutHome.path == "/" {
```

Comment:
I think that what you have should work on all unix platforms, I think you could probably remove the OS conditional and have the same code-path on all and this would compile everywhere. (there is the problem with case sensitive filesystems though). The more I think about this it would be simpler of we can remove the conditional and have the same behaviour and code everywhere that it would work, not sure what your opinion would be on doing that.

The exception being Windows as I am not sure how that works and if `~` is a valid path component or `HOME` would be set.

Swiftenv is tested on Windows (https://github.com/kylef/swiftenv/commit/9c463927c4c2dad900f6a5a97052c6afd64c77ec) but there are unfortunately no binaries to be able to use Swift. I think that `~` may work on Cygwin though. For FreeBSD, there is no CI I found unfortunately (and also the binary problem).